contrib/inventory/stacki.py

```diff
@@ -0,0 +1,187 @@
+    auth_creds  = {'USERNAME': params['stacki_user'],
+                   'PASSWORD': params['stacki_password']}
+
+    client = requests.session()
+    client.get(endpoint)
+
+    init_csrf = client.cookies['csrftoken']
+
+    header = {'csrftoken': init_csrf, 'X-CSRFToken': init_csrf,
+              'Content-type': 'application/x-www-form-urlencoded'}
+
+    login_endpoint = endpoint + "/login"
+
+    login_req = client.post(login_endpoint, data=auth_creds, headers=header)
+
+    csrftoken = login_req.cookies['csrftoken']
+    sessionid = login_req.cookies['sessionid']
+
+    auth_creds.update(CSRFTOKEN=csrftoken, SESSIONID=sessionid)
+
+    return client, auth_creds
+
+
+def stack_build_header(auth_creds):
+    header = {'csrftoken': auth_creds['CSRFTOKEN'],
+              'X-CSRFToken': auth_creds['CSRFTOKEN'],
+              'sessionid': auth_creds['SESSIONID'],
+              'Content-type': 'application/json'}
+
+    return header
+
+
+def stack_host_list(endpoint, header, client):
+
+    stack_r = client.post(endpoint, data=json.dumps({ "cmd": "list host"}),
+                          headers=header)
+    return json.loads(stack_r.json())
+
+
+def stack_net_list(endpoint, header, client):
+
+    stack_r = client.post(endpoint, data=json.dumps({ "cmd": "list host interface"}),
+                          headers=header)
+    return json.loads(stack_r.json())
+
+def format_meta(hostdata, intfdata, config):
+    use_hostnames = config['use_hostnames']
+    meta = dict(all=dict(hosts=list()),
+                frontends=dict(hosts=list()),
+                backends=dict(hosts=list()),
+                _meta=dict(hostvars=dict()))
+
+    # Iterate through list of dicts of hosts and remove
+    # environment key as it causes conflicts
+    for host in hostdata:
+        del host['environment']
+        meta['_meta']['hostvars'][host['host']] = host
+        meta['_meta']['hostvars'][host['host']]['interfaces'] = list()
+
+    for intf in intfdata:
+        if intf['host'] in meta['_meta']['hostvars']:
+            meta['_meta']['hostvars'][intf['host']]['interfaces'].append(intf)
+            if intf['default'] is True:
+                meta['_meta']['hostvars'][intf['host']]['ansible_host'] = intf['ip']
+                if not use_hostnames:
+                    meta['all']['hosts'].append(intf['ip'])
+                    if meta['_meta']['hostvars'][intf['host']]['appliance'] != 'frontend':
+                        meta['backends']['hosts'].append(intf['ip'])
+                    else:
+                        meta['frontends']['hosts'].append(intf['ip'])
+                else:
+                    meta['all']['hosts'].append(intf['host'])
+                    if meta['_meta']['hostvars'][intf['host']]['appliance'] != 'frontend':
+                        meta['backends']['hosts'].append(intf['host'])
+                    else:
+                        meta['frontends']['hosts'].append(intf['host'])
+    return meta
```

Comment:
This block would be a bit more legible, if you checked use_hostnames, then assigned to a temp variable whichever label the user has requested, and then appended that.

Something like (pseudopython):
```
this_host = host
if not use_hostnames:
    this_host = ip
if frontend:
    frontends.append(this_host)
else:
    backends.append(this_host)
```

This is admittedly a style thing, but I think it would be more readable/shorten a few lines.